plugins/modules/dataplatform_nodepool.py

```diff
@@ -0,0 +1,582 @@
+        'aliases': ['subscription_user'],
+        'env_fallback': 'IONOS_USERNAME',
+        'available': STATES,
+        'type': 'str',
+    },
+    'password': {
+        # Required if no token, checked manually
+        'description': ['The Ionos password. Overrides the IONOS_PASSWORD environment variable.'],
+        'aliases': ['subscription_password'],
+        'available': STATES,
+        'no_log': True,
+        'env_fallback': 'IONOS_PASSWORD',
+        'type': 'str',
+    },
+    'token': {
+        # If provided, then username and password no longer required
+        'description': ['The Ionos token. Overrides the IONOS_TOKEN environment variable.'],
+        'available': STATES,
+        'no_log': True,
+        'env_fallback': 'IONOS_TOKEN',
+        'type': 'str',
+    },
+    'wait': {
+        'description': ['Wait for the resource to be created before returning.'],
+        'default': True,
+        'available': STATES,
+        'choices': [True, False],
+        'type': 'bool',
+    },
+    'wait_timeout': {
+        'description': ['How long before wait gives up, in seconds.'],
+        'default': 600,
+        'available': STATES,
+        'type': 'int',
+    },
+    'state': {
+        'description': ['Indicate desired state of the resource.'],
+        'default': 'present',
+        'choices': STATES,
+        'available': STATES,
+        'type': 'str',
+    },
+}
+
+def transform_for_documentation(val):
+    val['required'] = len(val.get('required', [])) == len(STATES) 
+    del val['available']
+    del val['type']
+    return val
+
+DOCUMENTATION = '''
+---
+module: dataplatform_nodepool
+short_description: Create or destroy Data Platform Nodepools
+description:
+     - This is a simple module that supports creating or removing Data Platform Nodepools.
+       This module has a dependency on ionoscloud_dataplatform >= 1.0.0
+version_added: "2.0"
+options:
+''' + '  ' + yaml.dump(yaml.safe_load(str({k: transform_for_documentation(v) for k, v in copy.deepcopy(OPTIONS).items()})), default_flow_style=False).replace('\n', '\n  ') + '''
+requirements:
+    - "python >= 2.6"
+    - "ionoscloud_dataplatform >= 1.0.0"
+author:
+    - "IONOS Cloud SDK Team <sdk-tooling@ionos.com>"
+'''
+
+EXAMPLE_PER_STATE = {
+  'present' : '''
+  - name: Create Data Platform nodepool
+    dataplatform_nodepool:
+      name: "{{ name }}"
+      cluster: "a0a65f51-4d3c-438c-9543-39a3d7668af3"
+      node_count: "1"
+      cpu_family: "AMD_OPTERON"
+      cores_count: "1"
+      ram_size: "2048"
+      availability_zone: "AUTO"
+      storage_type: "SSD"
+      storage_size: "100"
+  ''',
+  'update' : '''
+  - name: Update Data Platform nodepool
+    dataplatform_nodepool:
+      name: "{{ name }}"
+      cluster: "ed67d8b3-63c2-4abe-9bf0-073cee7739c9"
+      node_count: 1
+      cores_count: "1"
+      maintenance_window:
+        day_of_the_week: 'Tuesday'
+        time: '13:03:00'
+      auto_scaling:
```

Comment:
I think auto_scaling needs to be added to OPTIONS

Reply:
Done ✔️  Thank you!